### PR TITLE
Got rid of banner instruction workaround

### PR DESCRIPTION
**Pull request by VysotskiVadim**
### Description
we don't need this workaround anymore. Banner instruction should always be present.

### CODE REVIEW

**Comment by Guardiola31337** [comment]
Took this for a quick spin and it seems it's not working as expected 👀 

https://user-images.githubusercontent.com/1668582/164014181-41b8c7ed-dda9-4e84-a281-60a879e2f7a0.mp4

File: instrumentation-tests/src/main/res/raw/route_response_dc_very_short.json [hunk 1]
```diff
@@ -1 +1,425 @@
...ht_typical":77.419,"duration_typical":60.163,"duration":56.73,"distance":300.781,"driving_side":"right","weight":73.259,"mode":"driving","geometry":"gu~diAz_t|qCFwK@g`@CsPVmBGqDE_DUsYCqCHiFNoDLyB\\uD@O`@cCbAiGpBwI~Lkr@t@}D"},{"voiceInstructions":[],"intersections":[{"bearings":[290],"entry":[true],"in":0,"admin_index":0,"geometry_index":20,"location":[-77.028556,38.89507]}],"maneuver":{"type":"arrive","instruction":"Your destination is on the left.","modifier":"left","bearing_after":0,"bearing_before":110,"location":[-77.028556,38.89507]},"name":"Pennsylvania Avenue Northwest","weight_typical":0,"duration_typical":0,"duration":0,"distance":0,"driving_side":"right","weight":0,"mode":"driving","geometry":"{|}diAvkm|qC??"}],"distance":378.255,"summary":"14th Street Northwest, Pennsylvania Avenue Northwest"}],"geometry":"cj}diAh`t|qCsc@IoECFwK@g`@CsPVmBGqDE_DUsYCqCHiFNoDLyB\\uD@O`@cCbAiGpBwI~Lkr@t@}D","voiceLocale":"en-US"}],"waypoints":[{"distance":0.356,"name":"14th Street Northwest","location":[-77.031957,38.89477]},{"distance":4.33,"name":"Pennsylvania Avenue Northwest","location":[-77.028556,38.89507]}],"code":"Ok","uuid":"2GFx3d97xAnu4ypO3ziVb5yX1vd4i1JvbKOnLQ8HGhNGTNVrAbfZUw=="}
```
Comment (VysotskiVadim) [inline instrumentation-tests/src/main/res/raw/route_response_dc_very_short.json:1]: replaced by the same route which has banner instructions

**Comment by VysotskiVadim** [comment]
> Took this for a quick spin and it seems it's not working as expected 👀
> 
>  20220419_092247.mp4

Now it should work fine. I added an instrumentation test to make sure that feature works. @mapbox/navigation-android , PR is ready for another round of review 🎉

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/trip/session/MapboxTripSessionNoSetupTest.kt [hunk 2]
```diff
@@ -263,6 +265,61 @@ class MapboxTripSessionNoSetupTest {
             assertEquals(listOf("1", "1"), newVoiceInstructions)
         }
 
+    @Test
+    fun `banner instruction is transferred from NN to SDK status update`() =
+        coroutineRule.runBlockingTest {
+            // arrange
```
Comment (Guardiola31337) [inline libnavigation-core/src/test/java/com/mapbox/navigation/core/trip/session/MapboxTripSessionNoSetupTest.kt:271]: Minor thing: What about removing the `// arrange` `// act` `// assert` comments and just mark the Triple A with new lines between the 3 blocks?

**Review by Guardiola31337 — APPROVED** [review]
Took this for a spin and it seems it's working fine now 👌 

Thanks for adding test coverage around this ❤️

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/trip/session/MapboxTripSessionNoSetupTest.kt (hunk 2 above)
Comment (VysotskiVadim) [reply libnavigation-core/src/test/java/com/mapbox/navigation/core/trip/session/MapboxTripSessionNoSetupTest.kt:271]: The test has an empty line in act, maybe that's a sign that the act sections is too big 🤔 
I agree with you in general but for this specific test I think the comments work better 🙂

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt [hunk 3]
```diff
@@ -308,42 +306,27 @@ internal class MapboxTripSession(
                 return
             }
 
-            updateRouteProgressJob?.cancel()
-            updateRouteProgressJob = mainJobController.scope.launch(Dispatchers.Main.immediate) {
```
Comment (dzinad) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:312]: Is it OK that you removed updateRouteProgressJob assignment here? Now it is only cancelled, never assigned. We should either remove it completely or return the assignment.

Comment (Guardiola31337) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:312]: Great catch! This went unnoticed, we definitely need to remove `updateRouteProgressJob` altogether as it's not used anymore. @VysotskiVadim could you cut a quick follow up PR when you have a chance? 🙏

Comment (VysotskiVadim) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:312]: @dzinad , thank you! https://github.com/mapbox/mapbox-navigation-android/pull/6136

**Pull request merged**